python/glow/gwas/logistic_regression.py

```diff
@@ -0,0 +1,185 @@
+        offset_df: pd.DataFrame = pd.DataFrame({}),
+        # TODO: fallback is probably not the best name
+        fallback: str = 'none',  # TODO: Make approx-firth default
+        fit_intercept: bool = True,
+        values_column: str = 'values',
+        dt: type = np.float64) -> DataFrame:
+    '''
+    Uses logistic regression to test for association between genotypes and one or more binary
+    phenotypes. This is a distributed version of the method from regenie: 
+    https://www.biorxiv.org/content/10.1101/2020.06.19.162354v2
+
+    On the driver node, we fit a logistic regression model based on the covariates for each
+    phenotype. We broadcast the resulting residuals, gamma vectors
+    (where gamma is defined as y_hat * (1 - y_hat)), and (C.T gamma C)^-1 matrices. In each task,
+    we then adjust the new genotypes based on the null fit, perform a score test as a fast scan
+    for potentially significant variants, and then test variants with p values below a threshold
+    using a more selective, more expensive test.
+
+    Args:
+        genotype_df : Spark DataFrame containing genomic data
+        phenotype_df : Pandas DataFrame containing phenotypic data
+        covariate_df : An optional Pandas DataFrame containing covariates
+        offset_df : An optional Pandas DataFrame containing the phenotype offset. The actual phenotype used
+                    for linear regression is `phenotype_df` minus the appropriate offset. The `offset_df` may
+                    have one or two levels of indexing. If one level, the index should be the same as the `phenotype_df`.
+                    If two levels, the level 0 index should be the same as the `phenotype_df`, and the level 1 index
+                    should be the contig name. The two level index scheme allows for per-contig offsets like
+                    LOCO predictions from GloWGR.
+        fallback : Which test to use for variants that meet a significance threshold for the score test
+        fit_intercept : Whether or not to add an intercept column to the covariate DataFrame
+        values_column : A column name or column expression to test with linear regression. If a column name is provided,
+                        `genotype_df` should have a column with this name and a numeric array type. If a column expression
+                        is provided, the expression should return a numeric array type.
+        dt : The numpy datatype to use in the linear regression test. Must be `np.float32` or `np.float64`.
+    '''
+
+    gwas_fx._check_spark_version(genotype_df.sql_ctx.sparkSession)
+    gwas_fx._validate_covariates_and_phenotypes(covariate_df, phenotype_df, is_binary=True)
+    sql_type = gwas_fx._regression_sql_type(dt)
+    genotype_df = gwas_fx._prepare_genotype_df(genotype_df, values_column, sql_type)
+    result_fields = [
+        # TODO: Probably want to put effect size and stderr here for approx-firth
```

Comment:
I think we can still calculate effect size and stderr without the corrections, right? As in: https://github.com/rgcgithub/regenie/blob/247483cd5617f048682062553265837c2b95d6ee/src/Data.cpp#L2456

Reply:
I saw that they compute `bhat`, but I don't understand what they actually represent. Based on the regenie paper and other resources, it seems that "effect size" universally corresponds to the maximum likelihood coefficient for the genotype feature. If that's the case, how could you know the effect size without fitting a model?

Reply:
I believe that effect size here simply refers to the difference between means, which is similar to the t-test stat (without the scaling).